docs/zetacore/gas_fee.md

```diff
@@ -0,0 +1,114 @@
+# ZetaChain Gas Fee Documentation
+
+## Overview
+
+Gas fees in ZetaChain refer to the amount paid by users to execute messages on connected chains. These fees **do not** include any gas payments that users make on the source chain to initiate transactions.
+
+## Key Concepts
+
+- **Deposit**: Transfer tokens from a connected chain to ZetaChain
+- **Withdraw**: Transfer tokens from ZetaChain to a connected chain
+- **Coin Types**: Different token types used for transactions (Gas, ERC20, Zeta)
+- **Revert**: When a transaction fails and needs to be reversed
+
+## V2 Protocol Flows
+
+### Deposit Transactions
+
+Deposits transfer tokens from connected chains to ZetaChain.
+
+- Gas Fee is always in GAS ZRC20. If the amount is in a different ERC20, it's swapped to buy Gas tokens
+
+| Transaction Type | GAS FEE Deposit | GAS FEE Revert | Notes |
+|------------------|-----------------|----------------|-------|
+| Deposit | Free | User Pays based on cointype | Revert fees paid using the GAS ZRC20 on source chain |
+| DepositAndCall | Free | User Pays based on cointype | Revert fees paid using the GAS ZRC20 on source chain |
+| Call | Free | Revert Not supported | Revert fees paid using the GAS ZRC20 on source chain |
+
+#### Gas Fee Payment by Coin Type
+
+1. **CoinType ERC20**
+   - User is using ERC20/ZRC20 tokens
+   - `Protocol` calculates expected gas cost on connected chain, using `Current Gas Price` * `Revert Gas Limit` + Protocol Flat Fee
+   - ERC20 tokens are swapped to buy gas ZRC20 tokens
+   - Gas ZRC20 tokens are burned to pay for the gas fee
+
+2. **CoinType Gas**
+   - User paid in the native gas token on a connected chain
+   - `Protocol` calculates expected gas cost on connected chain, using `Current Gas Price` * `Revert Gas Limit` + Protocol Flat Fee
+   - Gas token can be directly burned to pay for the gas fee
+
+3. **NoAssetCall**
+   - No revert processing in this case
+
+### Withdraw Transactions
+
+Withdrawals transfer tokens from ZetaChain to connected chains.
+
+| Transaction Type | GAS FEE Withdraw | GAS FEE Revert | Notes |
+|------------------|-----------------|----------------|-------|
+| Withdraw | User Pays based on withdraw type | Free call to OnRevert using fixed GasLimit | Initiated through Gateway smart contract |
+| WithdrawAndCall | User Pays based on withdraw type | Free call to OnRevert using fixed GasLimit | Initiated through Gateway smart contract |
+| Call | User Pays based on withdraw type | Free call to OnRevert using fixed GasLimit | Initiated through Gateway smart contract |
+
+- Withdrawals are initiated by users through zEVM smart contract calls
+- Fees are always paid in the GAS ZRC20 token for the target connected chain
```

Comment:
We should precise that unlike reverts, the amount is not deducted from the cctx funds, it is deducted directly from the sender balance